File: ios/Framework/Themes/Default/FileDisplay/ImageDisplayView_default.swift
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */
import UIKit


public class ImageDisplayView_default: BaseScreenletView, ImageDisplayViewModel {

	@IBOutlet public weak var imageView: UIImageView?

	public var imageMode: UIViewContentMode = .ScaleAspectFit

	public var placeholderImageMode: UIViewContentMode = .Center
	public var placeholder: UIImage?

	public var url: NSURL? {
		didSet {
			if let url = url, imageData = NSData(contentsOfURL: url) {
				imageView?.contentMode = self.imageMode
				imageView?.image = UIImage(data: imageData)
			}
		}
	}

	public var title: String? {
		didSet {
			self.presentingViewController?.title = title
		}
	}

	public override func onStartInteraction() {
		imageView?.contentMode = placeholderImageMode
		imageView?.image = placeholder
	}
}

File: xamarin/Framework/LiferayScreens.iOS/Themes/Default/FileDisplay/ImageDisplayView_default.cs
using CoreGraphics;
using Foundation;
using ObjCRuntime;
using System;
using UIKit;

namespace LiferayScreens
{
    // @interface ImageDisplayView_default : BaseScreenletView <ImageDisplayViewModel, FileDisplayViewModel>
    [BaseType(typeof(BaseScreenletView))]
    interface ImageDisplayView_default : IImageDisplayViewModel, IFileDisplayViewModel
    {
        // @property (nonatomic, weak) UIImageView * _Nullable imageView __attribute__((iboutlet));
        [NullAllowed, Export("imageView", ArgumentSemantic.Weak)]
        UIImageView ImageView { get; set; }

        // @property (nonatomic) UIViewContentMode imageMode;
        [Export("imageMode", ArgumentSemantic.Assign)]
        UIViewContentMode ImageMode { get; set; }

        // @property (nonatomic) UIViewContentMode placeholderImageMode;
        [Export("placeholderImageMode", ArgumentSemantic.Assign)]
        UIViewContentMode PlaceholderImageMode { get; set; }

        // @property (nonatomic, strong) UIImage * _Nullable placeholder;
        [NullAllowed, Export("placeholder", ArgumentSemantic.Strong)]
        UIImage Placeholder { get; set; }

        // @property (copy, nonatomic) NSURL * _Nullable url;
        [NullAllowed, Export("url", ArgumentSemantic.Copy)]
        NSUrl Url { get; set; }

        // @property (copy, nonatomic) NSString * _Nullable title;
        [NullAllowed, Export("title")]
        string Title { get; set; }

        // -(void)onStartInteraction;
        [Export("onStartInteraction")]
        void OnStartInteraction();

        // -(instancetype _Nonnull)initWithFrame:(CGRect)frame __attribute__((objc_designated_initializer));
        [Export("initWithFrame:")]
        [DesignatedInitializer]
        IntPtr Constructor(CGRect frame);
    }
}
